int64 right_count
int64 left_count
float32 delta_time
float32 degree